doc-changelog/parse_pr_title.py

```diff
@@ -0,0 +1,345 @@
+    }
+
+    # Save the changelog section to the CHANGELOG_SECTION environment variable
+    save_env_variable(
+        "CHANGELOG_SECTION", get_changelog_section(pr_labels, existing_labels)
+    )
+
+
+def get_changelog_section(pr_labels: dict, existing_labels: list) -> str:
+    """Find the changelog section corresponding to the label in the pull request.
+
+    Parameters
+    ----------
+    pr_labels: dict
+        Dictionary containing pull request labels and their corresponding changelog sections.
+    existing_labels: list
+        List of the labels that are in the pull request.
+
+    Returns
+    -------
+    str
+        The changelog section.
+    """
+    changelog_section = ""
+
+    # For each label key and changelog section value
+    for key, value in pr_labels.items():
+        # If the label is in the existing_labels list
+        if key in existing_labels:
+            # Save the changelog section based on the label
+            changelog_section = value
+            return changelog_section
+
+    # If no labels are in the PR, it goes into the miscellaneous category
+    changelog_section = "miscellaneous"
+    return changelog_section
+
+
+def clean_pr_title(pr_title: str, use_cc: str):
+    """Clean the pull request title.
+
+    Parameters
+    ----------
+    pr_title: str
+        The pull request title.
+    use_cc: str
+        Whether or not to use conventional commits to get the changelog section.
+    """
+    # Retrieve title
+    clean_title = pr_title
+
+    # If using conventional commits, remove it from title
+    if use_cc:
+        colon_index = clean_title.index(":")
+        clean_title = clean_title[colon_index + 1 :]
+
+    # Remove extra whitespace
+    clean_title = clean_title.strip()
+
+    # Add backslash in front of backtick and double quote
+    clean_title = clean_title.replace("`", "\\`").replace('"', '\\"')
+
+    # Save the clean pull request title as the CLEAN_TITLE environment variable
+    save_env_variable("CLEAN_TITLE", clean_title)
+
+
+def add_towncrier_config(org_name: str, repo_name: str, default_config: bool):
+    """Append the missing towncrier information to the pyproject.toml file.
+
+    Parameters
+    ----------
+    org_name: str
+        The name of the organization.
+    repo_name: str
+        The name of the repository.
+    default_config: bool
+        Whether or not to use the default towncrier configuration for the pyproject.toml file.
+    """
+    # Load pyproject.toml file
+    with open("pyproject.toml", "a+") as file:
+        config = toml.load("pyproject.toml")
+        tool = config.get("tool", "DNE")
+        towncrier = tool.get("towncrier", "DNE")
+
+        # List containing changelog sections under each release
+        changelog_sections = [
+            "added",
+            "dependencies",
+            "documentation",
+            "fixed",
+            "maintenance",
+            "miscellaneous",
+            "test",
+        ]
```

Comment:
These changelog sections all have to correspond to a [[tool.towncrier.type]] section, so even if someone says they don't want to use the default towncrier configuration, we still have to add the missing [[tool.towncrier.type]] sections so it doesn't fail to create the fragment.

If we want people to be able to define their own [[tool.towncrier.type]] sections, then I could read in these sections from their pyproject.toml and generate the changelog_sections list from that information. This would change the logic later in the script a little bit. Lmk what you think